lib/ansible/modules/cloud/azure/azure_rm_image_facts.py

```diff
@@ -0,0 +1,300 @@
+            returned: always
+            type: str
+        resource_group:
+            description:
+                - Resource group of the image.
+            returned: always
+            type: str
+        location:
+            description:
+                - Location of the image.
+            returned: always
+            type: str
+        os_disk:
+            description:
+                - Id of os disk for image.
+            type: str
+            sample: /subscriptions/xxxx/resourceGroups/xxx/providers/Microsoft.Compute/disks/xx
+        os_disk_caching:
+            description:
+                - Specifies caching requirements for the image.
+            returned: always
+            type: str
+        os_state:
+            description:
+                - Specifies image operating system state. Possible values are 'Generalized' or 'Specialized'.
+            returned: always
+            type: str
+            sample: Generalized
+        os_storage_account_type:
+            description:
+                - Specifies the storage account type for the managed disk.
+            type: str
+            sample: Standard_LRS
+        os_type:
+            description:
+                - Type of OS for image.
+            returned: always
+            type: str
+            sample: Linux
+        provisioning_state:
+            description:
+                - State of image.
+            returned: always
+            type: str
+            sample: Succeeded
+        source:
+            description:
+                - Resource id of source VM from which the image is created
+            type: str
+            sample: /subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.Compute/virtualMachines/xx
+        tags:
+            description:
+                - Dictionary of tags associated with the image.
+            type: complex
+        data_disks:
+            description:
+                - List of data disks associated with the image.
+            type: complex
+            returned: always
+            contains:
+                caching:
+                    description:
+                        - Type of caching of data disk.
+                    sample: read_only
+                disk_size_gb:
+                    description:
+                        - Specifies the size of empty data disks in gigabytes.
+                    returned: always
+                    type: int
+                    sample: 50
+                lun:
+                    description:
+                        - Specifies the logical unit number of the data disk.
+                    returned: always
+                    type: int
+                    sample: 0
+                storage_account_type:
+                    description:
+                        - Specifies the storage account type for the managed disk data disk.
+                    type: str
+                    sample: Standard_LRS
+                managed_disk:
```

Comment:
could we just have managed_disk_id? seems like a structure with a single field...

Reply:
Agree with you @zikalino . But it seems it can be an overhead with further processing, by iterating over all data_disks. Please let me know if it is feasible to do this.

Reply:
yes, it's ok to iterate and change, as long as users will get simpler structure :-)

Reply:
done with the changes. Please review @zikalino .
Thanks!